internal/webhooks/webhooks.go

```diff
@@ -13,18 +13,25 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
+var (
+	hncNamespace = ""
+	once         sync.Once
+)
+
+func SetHncNamespace(ns string) {
+	once.Do(func() {
+		hncNamespace = ns
+	})
+}
+
 // IsHNCServiceAccount is inspired by isGKServiceAccount from open-policy-agent/gatekeeper.
 func IsHNCServiceAccount(user *authnv1.UserInfo) bool {
 	if user == nil {
 		// useful for unit tests
 		return false
 	}
 
-	ns, found := os.LookupEnv("POD_NAMESPACE")
-	if !found {
-		ns = "hnc-system"
-	}
-	saGroup := fmt.Sprintf("system:serviceaccounts:%s", ns)
+	saGroup := fmt.Sprintf("system:serviceaccounts:%s", hncNamespace)
```

Comment:
I'm not sure when to use `IsHNCServiceAccount`, but if this func is called before `SetHncNamespace`, hncNamespace will be empty, it seems to not match expect.

Reply:
SetHncNamespace() is called in parseFlags() which is the first line in main().